src/librbd/mirror/snapshot/CreatePrimaryRequest.cc

```diff
@@ -205,15 +209,14 @@ void CreatePrimaryRequest<I>::unlink_peer() {
         unlink_snap_id = 0;
         continue;
       }
-      // call UnlinkPeerRequest only if the snapshot is linked with this peer
-      // or if it's not linked with any peer (happens if mirroring is enabled
-      // on a pool with no peers configured or if UnlinkPeerRequest gets
-      // interrupted)
-      if (info->mirror_peer_uuids.size() == 0) {
+      // call UnlinkPeerRequest if snapshot is not linked with any peer or
+      // incase if snapshot is incomplete
```

Comment:
This comment edit drops useful information ("happens if mirroring is enabled on a pool with no peers configured or if UnlinkPeerRequest gets interrupted").

Actually, this reminded me about something from last week.  Unrelated to the "skip" issue, I think we should be more careful here and act only on those incomplete snapshots that are linked with the peer at hand (and leave the comment intact since it would remain very much on point).  Something like this:
```
      // call UnlinkPeerRequest only if the snapshot is linked with this peer
      // or if it's not linked with any peer (happens if mirroring is enabled
      // on a pool with no peers configured or if UnlinkPeerRequest gets
      // interrupted)
      if (!info->mirror_peer_uuids.empty() &&
          info->mirror_peer_uuids.count(peer) == 0) {
        continue;
      }
      if (info->mirror_peer_uuids.empty() || !info->complete) {
        peer_uuid = peer;
        snap_id = snap_it.first;
        break;
      }
      count++;
```

Reply:
Sure, makesense!